src/js/api/api.js

```diff
@@ -309,43 +283,11 @@ define([
             });
         };
 
-
-        _this.registerPlugin = function (id, target, arg1, arg2) {
-            plugins.registerPlugin(id, target, arg1, arg2);
-        };
-
         function _forwardStateEvent(evt) {
             _this.trigger(evt.newstate, evt);
         }
 
-        _this.detachMedia = function () {
-            return _callInternal('jwDetachMedia');
-        };
-
-        _this.attachMedia = function (seekable) {
-            return _callInternal('jwAttachMedia', seekable);
-        };
-
-
-        _this.getAudioTracks = function () {
-            return _callInternal('jwGetAudioTracks');
-        };
-
-        function _callInternal() {
-            if (_playerReady) {
-                var args = Array.prototype.slice.call(arguments, 0),
-                    funcName = args.shift();
-                if (!_.isFunction(_controller[funcName])) {
-                    return null;
-                }
-                return _controller[funcName].apply(_controller, args);
-            }
-            _queuedCalls.push(arguments);
-        }
-
-        _this.callInternal = _callInternal;
```

Comment:
add it to deprecate. We can have a shoddy implementation or throw an exception / console.error when used.